From 837af51a9299510adae7e9b6b6be072b31d4535b Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Thu, 2 Nov 2017 00:49:05 +0000
Subject: [PATCH 3/8] nxp: NativeNfcManager: Implement missing inherited
 abstract methods

 * Report doGetNciVersion as default NCI_VERSION_1_0

 * Implement inactive doEnableDtaMode, doDisableDtaMode,
    and doSetScreenState native functions

Change-Id: Idf3892205ebe2bdefc01f781bbbc3e666f3febc3
---
 nxp/jni/com_android_nfc_NativeNfcManager.cpp  | 35 ++++++++++++++++---
 .../android/nfc/dhimpl/NativeNfcManager.java  | 30 +++++++++++++---
 2 files changed, 56 insertions(+), 9 deletions(-)

diff --git a/nxp/jni/com_android_nfc_NativeNfcManager.cpp b/nxp/jni/com_android_nfc_NativeNfcManager.cpp
index 22b92bcf..46972f1d 100644
--- a/nxp/jni/com_android_nfc_NativeNfcManager.cpp
+++ b/nxp/jni/com_android_nfc_NativeNfcManager.cpp
@@ -2356,11 +2356,26 @@ static jboolean com_android_nfc_NfcManager_doDownload(JNIEnv *e, jobject o)
     return performDownload(nat, true);
 }
 
-static jstring com_android_nfc_NfcManager_doDump(JNIEnv *e, jobject)
+static void com_android_nfc_NfcManager_doSetScreenState(JNIEnv*, jobject, jint screen_state_mask)
+{
+    ALOGD("%s: screen_state_mask = %d", __FUNCTION__, screen_state_mask);
+}
+
+static void com_android_nfc_NfcManager_doDump(JNIEnv*, jobject, jobject)
+{
+}
+
+static jint com_android_nfc_NfcManager_doGetNciVersion(JNIEnv*, jobject)
+{
+    return 0x10; /* NCI_VERSION_1_0 */
+}
+
+static void com_android_nfc_NfcManager_doEnableDtaMode(JNIEnv*, jobject)
+{
+}
+
+static void com_android_nfc_NfcManager_doDisableDtaMode(JNIEnv*, jobject)
 {
-    char buffer[100];
-    snprintf(buffer, sizeof(buffer), "libnfc llc error_count=%u", libnfc_llc_error_count);
-    return e->NewStringUTF(buffer);
 }
 
 /*
@@ -2422,8 +2437,18 @@ static JNINativeMethod gMethods[] =
    {"doSetP2pTargetModes","(I)V",
       (void *)com_android_nfc_NfcManager_doSetP2pTargetModes},
 
-   {"doDump", "()Ljava/lang/String;",
+   {"doSetScreenState", "(I)V",
+      (void *)com_android_nfc_NfcManager_doSetScreenState},
+
+   {"doDump", "(Ljava/io/FileDescriptor;)V",
       (void *)com_android_nfc_NfcManager_doDump},
+
+   {"getNciVersion","()I",
+      (void *)com_android_nfc_NfcManager_doGetNciVersion},
+   {"doEnableDtaMode", "()V",
+      (void *)com_android_nfc_NfcManager_doEnableDtaMode},
+   {"doDisableDtaMode", "()V",
+      (void *)com_android_nfc_NfcManager_doDisableDtaMode}
 };
 
 
diff --git a/nxp/src/com/android/nfc/dhimpl/NativeNfcManager.java b/nxp/src/com/android/nfc/dhimpl/NativeNfcManager.java
index 1a9b5fe0..4bd73991 100755
--- a/nxp/src/com/android/nfc/dhimpl/NativeNfcManager.java
+++ b/nxp/src/com/android/nfc/dhimpl/NativeNfcManager.java
@@ -30,6 +30,7 @@ import android.util.Log;
 import com.android.nfc.NfcDiscoveryParameters;
 
 import java.io.File;
+import java.io.FileDescriptor;
 
 /**
  * Native interface to the NFC Manager functions
@@ -122,6 +123,21 @@ public class NativeNfcManager implements DeviceHost {
         return doInitialize();
     }
 
+    private native void doEnableDtaMode();
+
+    @Override
+    public void enableDtaMode() {
+        doEnableDtaMode();
+    }
+
+    private native void doDisableDtaMode();
+
+    @Override
+    public void disableDtaMode() {
+        Log.d(TAG,"disableDtaMode : entry");
+        doDisableDtaMode();
+    }
+
     private native boolean doDeinitialize();
 
     @Override
@@ -140,7 +156,7 @@ public class NativeNfcManager implements DeviceHost {
     }
 
     @Override
-    public boolean routeAid(byte[] aid, int route) {
+    public boolean routeAid(byte[] aid, int route, int aidInfo) {
         return false;
     }
 
@@ -174,6 +190,12 @@ public class NativeNfcManager implements DeviceHost {
         return 0;
     }
 
+    @Override
+    public native void doSetScreenState(int screen_state_mask);
+
+    @Override
+    public native int getNciVersion();
+
     private native void doEnableDiscovery(int techMask,
                                           boolean enableLowPowerPolling,
                                           boolean enableReaderMode,
@@ -364,10 +386,10 @@ public class NativeNfcManager implements DeviceHost {
         return DEFAULT_LLCP_RWSIZE;
     }
 
-    private native String doDump();
+    private native void doDump(FileDescriptor fd);
     @Override
-    public String dump() {
-        return doDump();
+    public void dump(FileDescriptor fd) {
+        doDump(fd);
     }
 
     /**
-- 
2.17.1

